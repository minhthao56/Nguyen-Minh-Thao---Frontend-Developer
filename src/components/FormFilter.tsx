import Input from "./Input";
import PrimaryButton from "./PrimaryButton";

interface FormFilterProps {
  value?: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  onSearch?: React.MouseEventHandler<HTMLButtonElement>;
}

export default function FormFilter({
  value,
  onChange,
  onSearch,
}: FormFilterProps) {
  return (
    <div className="m-4">
      <h1 className="text-2xl uppercase mb-6">Search SpaceX</h1>
      <div className="flex">
        <Input
          type="text"
          placeholder="Search By Name"
          value={value}
          onChange={onChange}
        />
        <PrimaryButton type="submit" className="ml-5" onClick={onSearch}>
          Search
        </PrimaryButton>
      </div>
    </div>
  );
}
