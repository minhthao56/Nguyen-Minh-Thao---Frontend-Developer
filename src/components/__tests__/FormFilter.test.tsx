import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import FormFilter from "../FormFilter";

describe("<FormFilter/>", () => {
  const onChange = jest.fn();
  const onSearch = jest.fn();

  const renderComponent = () => {
    return render(<FormFilter onChange={onChange} onSearch={onSearch} />);
  };

  it("should match to snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should call onChange when typing search input", () => {
    renderComponent();

    userEvent.type(screen.getByPlaceholderText("Search By Name"), "test");

    expect(onChange).toBeCalled();
  });

  it("should call onSearch when click button search", () => {
    renderComponent();

    userEvent.click(screen.getByRole("button", { name: "Search" }));

    expect(onSearch).toBeCalled();
  });
});
