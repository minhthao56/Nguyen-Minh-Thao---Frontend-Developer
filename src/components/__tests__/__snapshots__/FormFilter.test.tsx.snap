// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFilter/> should match to snapshot 1`] = `
<div>
  <div
    class="m-4"
  >
    <h1
      class="text-2xl uppercase mb-6"
    >
      Search SpaceX
    </h1>
    <div
      class="flex"
    >
      <input
        class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-black focus:black block w-full p-2.5 undefined"
        placeholder="Search By Name"
        type="text"
        value=""
      />
      <button
        class="text-white bg-black focus:ring-4 focus:outline-none focus:ring-black font-medium rounded-lg text-sm w-full px-5 py-2.5 text-center ml-5"
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
